plugins/hls-module-name-plugin/src/Ide/Plugin/ModuleName.hs

```diff
@@ -160,9 +167,11 @@ pathModuleNames recorder state normFilePath filePath
 -- | The module name, as stated in the module
 codeModuleName :: IdeState -> NormalizedFilePath -> IO (Maybe (Range, T.Text))
 codeModuleName state nfp = runMaybeT $ do
-  pm <- MaybeT . runAction "ModuleName.GetParsedModule" state $ use GetParsedModule nfp
+  (pm, mp) <- MaybeT . runAction "ModuleName.GetParsedModule" state $ useWithStale GetParsedModule nfp
   L (locA -> (RealSrcSpan l _)) m <- MaybeT . pure . hsmodName . unLoc $ pm_parsed_source pm
-  pure (realSrcSpanToRange l, T.pack $ moduleNameString m)
+  let range = realSrcSpanToRange l
+  let range' = fromMaybe range (toCurrentRange mp range)
```

Comment:
This seems wrong, shouldn't we fail or give up somehow if we can't map the range to the current thing, not use the probably-wrong range we have?

Reply:
Turns out I can't have a counterexample that `toCurrentRange` can fail, keep it unless we know how it fails.

Reply:
Seems risky to me :shrug: Why does it return a `Maybe` if it can't happen? And if it does happen we'll just silently do the wrong thing.

Reply:
Well, it makes sense, I'll let them be failed if `toCurrentRange` returns `Nothing`. But I'm still curious about how it fails 😅

Reply:
If you find out, please add some Haddock! :upside_down_face: